Libraries/WebSocket/RCTWebSocketModule.m

```diff
@@ -44,9 +44,9 @@ - (void)dealloc
   }
 }
 
-RCT_EXPORT_METHOD(connect:(NSURL *)URL protocols:(NSArray *)protocols options:(NSDictionary *)options socketID:(nonnull NSNumber *)socketID)
+RCT_EXPORT_METHOD(connect:(NSURL *)URL protocols:(NSArray *)protocols headers:(NSDictionary *)headers socketID:(nonnull NSNumber *)socketID)
 {
-  RCTSRWebSocket *webSocket = [[RCTSRWebSocket alloc] initWithURL:URL protocols:protocols options:options];
+  RCTSRWebSocket *webSocket = [[RCTSRWebSocket alloc] initWithURL:URL protocols:protocols headers:headers];
```

Comment:
We can just merge the url and headers into an NSURLRequest at this point, and use the other constructor.

    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
    request.allHTTPHeaderFields = headers;